Windows Task Scheduler Watchdog

RchGrav opened the pull request:
Added the ability to have the utility start non-interactively, restarts get-my-ps5 if for any reason it stops.

Running taskmaster.bat once creates a 5 minute watchdog to restart get-my-ps5 if it exits prematurely..

Running it a second time kills get-my-ps5 and the 5 minute watchdog task.

I'm hesitant to say include this in the main repo because it may make setting this up a little too easy.  I recommend making a shortcut to this so you can easily kill the watchdog once you are in the queue, you don't want it to fire up a second space in the queue I would think.

RchGrav committed:
Create setup_task.bat

RchGrav committed:
Update index.js

RchGrav committed:
Create toggle_task.bat

RchGrav committed:
Delete setup_task.bat

RchGrav committed:
Rename toggle_task.bat to taskmaster.bat

cpolitz commented:
This is a good idea but might be better accomplished using nssm (https://nssm.cc/).  This would set it up as a windows service instead of a task and would auto run on start up.  Don't know for sure just making a suggestion.

RchGrav commented:
It was mainly so it autostarts if I reboot, etc...  it checks to see if its running and restarts it if not...  before when the site would have a queue w/ false alarms at odd hours or timeout.  I assume those are all fixed?

cpolitz commented:
ah gotcha, no I think those are still some persistent issues.  It seems
real queues only occur between 2pm - 6pm central.

On Sun, Dec 13, 2020 at 10:35 AM Richard Graver <notifications@github.com>
wrote:

> It was mainly so it autostarts if I reboot, etc... it checks to see if its
> running and restarts it if not... before when the site would have a queue
> w/ false alarms at odd hours or timeout. I assume those are all fixed?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/camtheperson/get-my-ps5/pull/44#issuecomment-744033499>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD6XYG7PROT4XYXHCZBRX5DSUTULRANCNFSM4UZD6FFA>
> .
>


-- 
Thanks,

Chris

RchGrav commented:
This is running interactively just as normal..    I don't like the idea of it running in the background. anything there isnt anything equiv to gnu screen for windows so you can watch the console.. telnet wrapper MAYBE.. anyway.. watching the console is important I think.. if Sony bans your IP you could mitigate and get a new IP by changing your mac address.. or whatever..   Also if it was fully backgrounded as a service I'm not sure if child processed become interactive...

It works as a toggle.. run it once its kept running,, run it again process killed and so is the task..  even if the code in the main app is solid there is always something.. maybe someone closes the window... ?!     Its just a safety net../  try it if you are on a windows box.  ;-)    

I'm not necessarily saying add it to the repo...    at least until I finally get a PS5 for a gift.  ;-)

This solution does have a flaw in that I assume once the current version open the browser it exits...   nothing has told the watchdog to stop so it could theortically pop up and launch a second queue page.

RchGrav commented:
Regarding the 2-6pm...  I noticed the same thing looking though the history of the drops.   but is that REALLY a rule or just what has happened so far.

RchGrav commented:
Gonna keep a link for nssm tho.. nice find.   I've used srvany, it looks like a good replacement.

camtheperson commented on src/index.js:
```diff
@@ -1,6 +1,8 @@
 const promptly = require("promptly");
 const puppeteer = require("puppeteer");
+const choice = ("disc");
+const alarm = ("Y");
```
We shouldn't be setting these here. They should be based on user input.

camtheperson commented on src/index.js:
```diff
@@ -18,8 +20,8 @@ const playstationType = {
-    const choice = await promptly.choose("Which version would you like? (disc or digital)", ["disc", "digital"]);
-    const alarm = await promptly.choose("Would you like to hear a loud, annoying alarm when we find your PS5? (Y or N)", ["Y", "N"])
+    if (choice == "") choice = await promptly.choose("Which version would you like? (disc or digital)", ["disc", "digital"]);
+    if (alarm == "") alarm = await promptly.choose("Would you like to hear a loud, annoying alarm when we find your PS5? (Y or N)", ["Y", "N"])
```
Won't this always evaluate to false?

RchGrav commented:
can they be passed in as a variable? or ask would you like to save these
settings?

They can be set to "" and then they are based on user input....  I'll make
a change.

On Mon, Dec 14, 2020 at 1:53 PM Cameron Hermens <notifications@github.com>
wrote:

> *@camtheperson* commented on this pull request.
> ------------------------------
>
> In src/index.js
> <https://github.com/camtheperson/get-my-ps5/pull/44#discussion_r542641541>
> :
>
> > @@ -1,6 +1,8 @@
>  const open = require("open");
>  const promptly = require("promptly");
>  const puppeteer = require("puppeteer");
> +const choice = ("disc");
> +const alarm = ("Y");
>
> We shouldn't be setting these here. They should be based on user input.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/camtheperson/get-my-ps5/pull/44#pullrequestreview-551795816>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAE5BIMREHCXO63A5VTPDUTSUZNKPANCNFSM4UZD6FFA>
> .
>

RchGrav committed:
Update index.js

Allow user to choose how script starts interactive, or non-interactive.   (cron, task sceduler, NSSM etc compatibility)

camtheperson commented:
@RchGrav Just saw your note about being hesitant merging this in.

For now, I think this would best live as an experimental feature within your fork. If you get to a point where you can validate it through more testing, please consider opening another PR.

Closing this out. Thanks :)

RchGrav commented:
fair enough.. I concur
